src/Build/BackEnd/Shared/BuildResult.cs

```diff
@@ -496,7 +475,7 @@ internal bool ResultBelongsToRootRequest
         /// <returns>The results for the specified target</returns>
         /// <exception>KeyNotFoundException is returned if the specified target doesn't exist when reading this property.</exception>
         /// <exception>ArgumentException is returned if the specified target already has results.</exception>
-        public ITargetResult this[string target]
+        public TargetResult this[string target]
```

Comment:
Was this intended to be a breaking change? This seems to have broken our usage of the indexer. @AndyGerlicher